pkg/ingester/ingester_v2.go

```diff
@@ -129,6 +145,11 @@ func NewV2(cfg Config, clientConfig client.Config, limits *validation.Overrides,
 	i.done.Add(1)
 	go i.updateLoop()
 
+	if i.cfg.TSDBConfig.HeadCompactionConcurrency > 0 && i.cfg.TSDBConfig.HeadCompactionInterval > 0 {
```

Comment:
Why is it possible to disable compactions? We should never allow it. Even if you run without shipper you definitely wants compactions otherwise your memory will soon explode.

I would suggest:
1. Add a config validation to ensure that `HeadCompactionConcurrency > 0`
2. Always run `compactionLoop()`

What's your take?

Reply:
I left an option to disable it when running tests. But now I think it doesn't make sense even in that scenario. I'll modify it.